src/KakaoMap.js

```diff
@@ -13,11 +31,114 @@ class KakaoMap extends React.Component {
         this.state = {};
     }
 
+    createMouseClickListener = (map, customOverlay) => {
+        let overlay = false;
+        const toggle = () => overlay = !overlay;        
+        return () => {
+            toggle();
+            overlay ? customOverlay.setMap(map) : customOverlay.setMap(null);
+        }
+    }
+
+    wrapDetails = (details) => {
+        let result = [];
+        for (let detail in details) {
+            let temp = <Detail key={detail}>{detail} : {details[detail]}</Detail>;
+            result.push(temp);
+        }
+        return result;
+    }
+
+    createCustomOverlay = (map, marker, data) => {
+        const { 
+            iwBackgroundColor,
+            iwFontColor,
+            iwFontSize
+         } = this.props;
+
+        let content = <Content 
+                        iwBackgroundColor={iwBackgroundColor}
+                        iwFontColor={iwFontColor}
+                        iwFontSize={iwFontSize}
+                        >
+                        {this.wrapDetails(data.details)}
+                      </Content>;
+
+        let customOverlay = new kakao.maps.CustomOverlay({
+            content: renderToString(content),
+            position: marker.getPosition(),
+            xAnchor: 0.5,
+            yAnchor: 0
+        });
+
+        kakao.maps.event.addListener(
+            marker,
+            'click',
+            this.createMouseClickListener(map, customOverlay)
+        );
+    }
+
+    mapScript = () => { 
+
+        const { 
+            appKey,
+            latitude,
+            longitude,
+            level,
+            dataList
+         } = this.props;
+
+        const script = document.createElement("script");
+        script.async = true;
+        script.src = `https://dapi.kakao.com/v2/maps/sdk.js?appkey=${appKey}&autoload=false`;
+        document.head.appendChild(script);
+
+        script.onload = () => {
+            kakao.maps.load(() => {
+
+                let container = document.getElementById("map");
+                let options = {
+                    center: new kakao.maps.LatLng(latitude, longitude),
+                    level: level ? level : 3
+                };
+                const map = new window.kakao.maps.Map(container, options);
+
+                dataList.forEach((data) => {
+                    const marker = new kakao.maps.Marker({
+                        map: map,
+                        position: new kakao.maps.LatLng(data.latitude, data.longitude)
+                    });
+                    data.details ? this.createCustomOverlay(map, marker, data) : '';
+                });
+
+            });
+        };
+    };
+      
+    componentDidMount() {
+        this.mapScript();
+    }
```

Comment:
I think it would be better if the component lifecycle methods came after the constructor.